ros1_rosbag_storage_vendor/rosbag1_ament_patch.diff

```diff
@@ -0,0 +1,298 @@
++)
+ 
+ install(TARGETS rosbag_default_encryption_plugins
+-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
++  ARCHIVE DESTINATION lib
++  LIBRARY DESTINATION lib
++  RUNTIME DESTINATION bin
+ )
+ 
+ install(DIRECTORY include/
+-  DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION}
+-  FILES_MATCHING PATTERN "*.h"
++  DESTINATION include
+ )
+ 
+ if(NOT WIN32)
+   install(FILES encryptor_plugins.xml
+-    DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
++    DESTINATION share/
+   )
+ 
+-  if(CATKIN_ENABLE_TESTING)
+-    find_package(rostest)
+-
+-    catkin_add_gtest(test_aes_encryptor test/test_aes_encryptor.cpp src/gpgme_utils.cpp
+-      WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/test)
+-    if(TARGET test_aes_encryptor)
+-      target_link_libraries(test_aes_encryptor rosbag_storage ${BZIP2_LIBRARIES}  ${AES_ENCRYPT_LIBRARIES} ${catkin_LIBRARIES} ${Boost_LIBRARIES})
+-    endif()
+-  endif()
++#  if(BUILD_TESTING)
++#    find_package(ament_cmake_gmock)
++#
++#    catkin_add_gtest(test_aes_encryptor test/test_aes_encryptor.cpp
++#      WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/test)
++#    if(TARGET test_aes_encryptor)
++#      target_link_libraries(test_aes_encryptor rosbag_storage ${catkin_LIBRARIES} ${Boost_LIBRARIES})
++#    endif()
++#  endif()
+ endif()
++
++ament_export_include_directories(include)
++ament_export_libraries(ros1_rosbag_storage)
++ament_export_dependencies(
++  console_bridge
++  pluginlib
++  BZip2
++)
++
++ament_package()
+diff --git a/tools/rosbag_storage/encryptor_plugins.xml b/tools/rosbag_storage/encryptor_plugins.xml
+index 3128ec9a5..fb66f450c 100644
+--- a/tools/rosbag_storage/encryptor_plugins.xml
++++ b/tools/rosbag_storage/encryptor_plugins.xml
+@@ -1,4 +1,4 @@
+-<library path="lib/librosbag_default_encryption_plugins">
++<library path="rosbag_default_encryption_plugins">
+   <class name="rosbag/NoEncryptor" type="rosbag::NoEncryptor" base_class_type="rosbag::EncryptorBase">
+     <description>This is a plugin for no encryption.</description>
+   </class>
+diff --git a/tools/rosbag_storage/package.xml b/tools/rosbag_storage/package.xml
+index 502f5308b..3f28d00f1 100644
+--- a/tools/rosbag_storage/package.xml
++++ b/tools/rosbag_storage/package.xml
+@@ -1,5 +1,5 @@
+-<package>
+-  <name>rosbag_storage</name>
++<package format="2">
++  <name>ros1_rosbag_storage</name>
+   <version>1.14.3</version>
+   <description>
+     This is a set of tools for recording from and playing back ROS
+@@ -8,35 +8,25 @@
+   <maintainer email="dthomas@osrfoundation.org">Dirk Thomas</maintainer>
+   <license>BSD</license>
+ 
+-  <buildtool_depend>catkin</buildtool_depend>
++  <buildtool_depend>ament_cmake</buildtool_depend>
+ 
++  <build_depend>ros1_bridge</build_depend>
+   <build_depend>boost</build_depend>
+   <build_depend>bzip2</build_depend>
+-  <build_depend version_gte="0.3.17">cpp_common</build_depend>
```

Comment:
why aren't these deps necessary?

Reply:
This is similar to what the `ros1_bridge` does:

From my understanding from the discussion with Dirk Thomas on the ros1_bridge PR:
- We only `find_package`, `ament_export_dependencies` and put into the `package.xml` all dependencies to ros2 packages and non-ROS packages (like bzip2 and boost). 
- All ros1 packages are found by the `find_ros1_package`. This ensures that the dependency tree does not get confused between ros2 and ros1 packages, which don't play nice together.
- At the moment (no environment hooks present) this also means that all consuming downstream packages need to find those dependencies again. This is not nice but can be amended by registering the correct hooks to `ament_package` (however, I don't understand the process well enough to do this with limited time)